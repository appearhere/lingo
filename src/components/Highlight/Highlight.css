.mark {
  color: var(--color-white);
  background-color: var(--color-gold);
}
