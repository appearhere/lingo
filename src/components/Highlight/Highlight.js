import React from 'react';
import replace from 'string-replace-to-array';
import PropTypes from 'prop-types';

import css from './Highlight.css';

const Highlight = ({ highlight, caseSensitive, text, ...rest }) => (
  <div { ...rest }>
    { highlight
      ? replace(
        text,
        new RegExp(
          (highlight || '').replace(/[-\\^$*+?.()|[\]{}]/g, '\\$&'),
          caseSensitive ? 'g' : 'gi'
        ),
        (tag, index) => <mark key={ index } className={ css.mark }>{ tag }</mark>
      )
      : text }
  </div>
);


Highlight.propTypes = {
  highlight: PropTypes.string,
  text: PropTypes.string,
  caseSensitive: PropTypes.bool,
};

Highlight.defaultProps = {
  caseSensitive: false,
};

export default Highlight;
