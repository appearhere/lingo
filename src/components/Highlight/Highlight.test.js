import React from 'react';
import renderer from 'react-test-renderer';

import Highlight from './Highlight';

describe('Highlight component', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Highlight />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
