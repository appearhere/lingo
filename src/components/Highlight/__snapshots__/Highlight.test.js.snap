// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Highlight component renders correctly 1`] = `<div />`;
